app/classes/content_filter.rb

```diff
@@ -1,96 +1,117 @@
+# observation_filter_keys:: Keys for those observation filters.
+# observation_filters_with_checkboxes:: Observation filters with checkboxes.
 #
-# To be able to override the new filter in Advanced Searches, at least:
-#   Add tests, e.g., to TestAdvancedSearchFilters
-#   Supplement fixtures as needed by added tests
-#   In /config/locales/en.txt define text to be displayed before and next to
-#     radio boxes, using advanced_search_filter_has_images: as a model.
-# * Supplement _advanced_search_filters.html.erb as needed
-# * Supplement ObservationController#advanced_search_form as needed.
-# * Supplement ApplicationController#show_index_of_objects as needed.
-# * Supplement Query::RssLogBase as needed.
+# Instance methods:
 #
-# To also add it as a Pattern Search
-#   Supplement PatternSearchTest
-#   Supplement classes/pattern_search/observation.rb or -- if it's not an
-#     Observation search -- add a new pattern_search file and class.
+# sym::           Filter mame (symbol), same as Query param and User column.
+# model::         Model on which filter operates.
+# checkbox::      Name of account/prefs form checkbox (symbol).
+# on_vals::       Array of allowed values when filter is on, in the order you
+#                 want them to appear in advanced_filters
+# checked_val::   Value when checkbox is checked.
+# off_val::       Value when filter is off.
 #
-# There are probably other steps/files I've forgotten. JDC 2016-09-15
+class ContentFilter
 
-  ### filter definitions ###
-  # In the order you want filters to appear in advanced_search
-  # name:         filter name, as string
-  # sym:          filter mame, as symbol
-  # model:        model on which filter operates.  Used by ContentFilter
-  # checkbox:     prefs form checkbox, e.g., :has_images_checkbox.
-  #               Used by prefs view
-  # checked_val:  value when checkbox checked
-  # off_val:      value when filter is off
-  # on_vals:      array of allowed values when filter is on,
-  #               in order you want them to appear in advanced_filters
-  # sql_cond:     predicate added to Query "where", when filter is on.  E.g.:
-  #               "observations.specimen IS #{params[:has_specimen]}"
+  # To add a new Content Filter: (The leading asterisk (*) indicates
+  # an unnecessary step for new observation filter -- it's done
+  # automatically.)
+  #
+  #   Add tests, e.g., to TestUserContentFilter
+  #   Supplement fixtures as needed by added tests
+  #   Add a filter definition below, and add that definition to #filters below
+  #   Add a new checkbox_val method in User
+  #   In /config/locales/en.txt define text to be displayed before and next to
+  #     check boxes, using prefs_obs_filters_has_images: as a model.
+  # * Supplement _prefs_filters.html.erb as needed
+  # * Add filter to AccountController#prefs_types & #update_content_filter
+  # * For Observation filter, supplement Query::Initializers::ObservationFilters
```

Comment:
No longer done automatically. (Developer must add observation_filter_parameter_declarations manually.) So asterisk should be deleted.
```#   For Observation filter, supplement Query::Initializers::ObservationFilters```